src/Microsoft.DotNet.Darc/DarcLib/VirtualMonoRepo/GitNativeRepoCloner.cs

```diff
@@ -0,0 +1,106 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using Microsoft.DotNet.DarcLib.Helpers;
+using Microsoft.Extensions.Logging;
+
+#nullable enable
+namespace Microsoft.DotNet.DarcLib.VirtualMonoRepo;
+
+/// <summary>
+/// Class that clones repos by calling out to the native git executable.
+/// Does not uses Libgit2Sharp which is quite memory hungry when dealing with large repos.
+/// </summary>
+public class GitNativeRepoCloner : IGitRepoCloner
+{
+    private const string GitAuthUser = "dn-bot";
```

Comment:
If I'm syncing the VMR locally, does this attempt to always use dn-bot as the user to auth as? Does this work for AzDO?

Reply:
The user sort of doesn't matter when you use that PAT both for GH and AzDO so this is just something to put in there.

If you sync locally, you don't need to provide a PAT as the git credential manager will kick in and it should work without a PAT. I only tested in a docker container though as I knew I don't have any auth set up there to verify it works.